pandas/core/frame.py

```diff
@@ -2713,7 +2713,9 @@ def assign(self, **kwargs):
         8   9  0.549296  2.197225
         9  10 -0.758542  2.302585
         """
-        data = self.copy()
+
+        # See GH19163
+        data = self.copy().to_dense()
```

Comment:
Could you define assign on SparseDataFrame and only density if nescessary?

Reply:
and actually you don't want to densify, rather you want to do something like (in SparseDataFrame)

```
def assign(self, **kwargs):

      # coerce to a DataFrame
      self = DataFrame(self)
      return self.assign(**kwargs)
```

this actually ends up copying twice though. So the *real* solution is to move  the guts of ``DataFrame.assign``  to ``_assign`` (and leave the copy part in ``.assign``), then call ``._assign`` in the sparse version.